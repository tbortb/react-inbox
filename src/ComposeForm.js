import { Component } from 'react';

class ComposeForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            subject: "",
            body: ""
        }
    }

    updateNewMessage = e => {
        this.setState(prevState => ({
            [e.target.dataset.target]: e.target.value
        }))
    }

    submit = e => {
        this.props.addFunc({
            subject: this.state.subject,
            body: this.state.body
        });
        //clear state
        this.setState(prevState => ({ subject: "", body: "" }))
    }

    render() {
        return <form className="form-horizontal well" onSubmit={this.submit}>
            <div className="form-group">
                <div className="col-sm-8 col-sm-offset-2">
                    <h4>Compose Message</h4>
                </div>
            </div>
            <div className="form-group">
                <label htmlFor="subject" className="col-sm-2 control-label">Subject</label>
                <div className="col-sm-8">
                    <input type="text" className="form-control" id="subject" data-target="subject"
                        placeholder="Enter a subject" name="subject" onChange={this.updateNewMessage}>
                    </input>
                </div>
            </div>
            <div className="form-group">
                <label htmlFor="body" className="col-sm-2 control-label">Body</label>
                <div className="col-sm-8">
                    <textarea name="body" id="body" data-target="body"
                        className="form-control" onChange={this.updateNewMessage}></textarea>
                </div>
            </div>
            <div className="form-group">
                <div className="col-sm-8 col-sm-offset-2">
                    <input type="submit" value="Send" className="btn btn-primary">
                    </input>
                </div>
            </div>
        </form>
    }
}

export default ComposeForm