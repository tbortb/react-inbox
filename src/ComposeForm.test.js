import React from 'react';
import ComposeForm from './ComposeForm';
import renderer from 'react-test-renderer';

test('renders the same way every time', () => {
  const tree = renderer.create( <ComposeForm /> ).toJSON();
  expect(tree).toMatchSnapshot();
});