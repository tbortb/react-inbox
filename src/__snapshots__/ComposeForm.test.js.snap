// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same way every time 1`] = `
<form
  className="form-horizontal well"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="col-sm-8 col-sm-offset-2"
    >
      <h4>
        Compose Message
      </h4>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 control-label"
      htmlFor="subject"
    >
      Subject
    </label>
    <div
      className="col-sm-8"
    >
      <input
        className="form-control"
        data-target="subject"
        id="subject"
        name="subject"
        onChange={[Function]}
        placeholder="Enter a subject"
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 control-label"
      htmlFor="body"
    >
      Body
    </label>
    <div
      className="col-sm-8"
    >
      <textarea
        className="form-control"
        data-target="body"
        id="body"
        name="body"
        onChange={[Function]}
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="col-sm-8 col-sm-offset-2"
    >
      <input
        className="btn btn-primary"
        type="submit"
        value="Send"
      />
    </div>
  </div>
</form>
`;
